Source/WebGPU/WebGPU/CommandEncoder.swift

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021-2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+public func clearBuffer(
+    commandEncoder: WebGPU.CommandEncoder, buffer: WebGPU.Buffer, offset: UInt64, size: inout UInt64
+) {
+    commandEncoder.clearBuffer(buffer: buffer, offset: offset, size: &size)
+}
+
+extension WebGPU.CommandEncoder {
+    public func clearBuffer(buffer: WebGPU.Buffer, offset: UInt64, size: inout UInt64) {
+        guard self.prepareTheEncoderState() else {
+            self.generateInvalidEncoderStateError()
+            return
+        }
+        if size == UInt64.max {
+            let initialSize: UInt64 = buffer.initialSize()
+            let (subtractionResult, didOverflow) = initialSize.subtractingReportingOverflow(offset)
+            if didOverflow {
+                self.device().generateAValidationError(
+                    Swift.String("CommandEncoder::clearBuffer(): offset > buffer.size"))
```

Comment:
```suggestion
                    "CommandEncoder::clearBuffer(): offset > buffer.size")
```

does this not work?

Reply:
For some reason WTF.String has been imported here as String. So if I use the default then it becomes a WTF.String.

Reply:
even string literals in the Swift file get turned into WTF strings? I would not expect that, and we should definitely try to fix that if so

Reply:
fixed now. This is the culprit. 
https://github.com/WebKit/WebKit/blob/2509a2481fe05555bffdeb1afa21471afdc0fd7e/Source/WTF/wtf/text/WTFString.h#L559

Although, I think the swift compiler should stop that from happening.